lib/pricing-poller.js

```diff
@@ -136,44 +136,49 @@ class PricingPoller {
 
     for (let region of this.regions) {
       let ec2 = this.ec2[region];
-      let [azResult, priceResult] = await Promise.all([
-        this.runaws(ec2, 'describeAvailabilityZones', {
-          Filters: [{
-            Name: 'state',
-            Values: ['available'],
-          }],
-        }),
-        this.runaws(ec2, 'describeSpotPriceHistory', {
+      let nextToken = undefined;
+      let azResult = await this.runaws(ec2, 'describeAvailabilityZones', {
+        Filters: [{
+          Name: 'state',
+          Values: ['available'],
+        }],
+      });
+      let zones = azResult.AvailabilityZones.map(x => x.ZoneName);
+
+      do {
+        let priceResult = await this.runaws(ec2, 'describeSpotPriceHistory', {
           StartTime: startTime,
+          NextToken: nextToken,
           Filters: [{
             Name: 'product-description',
             Values: ['Linux/UNIX'],
           }],
-        }),
-      ]);
+        });
 
-      // Get the information from the API responses that we care about
-      let zones = azResult.AvailabilityZones.map(x => x.ZoneName);
-      let pricePoints = priceResult.SpotPriceHistory.filter(x => zones.includes(x.AvailabilityZone));
+        nextToken = priceResult.NextToken;
 
-      // Determine what the prices which the API described.
-      let spotPrices = this._findSpotPricesForRegion({pricePoints});
-      let onDemandPrices = this._findOnDemandPricesForRegion();
+        // Get the information from the API responses that we care about
+        let pricePoints = priceResult.SpotPriceHistory.filter(x => zones.includes(x.AvailabilityZone));
```

Comment:
heh, i couldn't figure out why you deleted `zones`, but it makes sense, since it was moved before the loop!